### PR TITLE
test: 10 tests from others

### PR DESCRIPTION
**Pull request by mintbomb27**
## Related Issue(s)
#546 

## Description
Contains 10 kuttl tests from others

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [ ] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: other/require-unique-external-dns/svc.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    external-dns.alpha.kubernetes.io/hostname: "1.1.1.1"
```
Comment (chipzoller) [inline other/require-unique-external-dns/svc.yaml:5]: These aren't host names but that's ok, it still works.

File: other/restart-deployment-on-secret-change/06-check-restart.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+- script: "if [ \"$(kubectl get pods -n default | grep busybox | awk 'NR==1{print $1}')\" != \"$(cat pod-name.txt)\" ];then exit;else exit 1;fi"
```
Comment (chipzoller) [inline other/restart-deployment-on-secret-change/06-check-restart.yaml:4]: You could have also compared the resourceVersion in the modified secret with the value of the `ops.corp.com/triggerrestart` but this is also ok :)

**Review by chipzoller — APPROVED** [review]
Nice

File: other/restrict-annotations/99-delete.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+delete:
+- apiVersion: kyverno.io/v1
+  kind: ClusterPolicy
+  name: require-unique-uid-per-workload
```
Comment (chipzoller) [inline other/restrict-annotations/99-delete.yaml:6]: Copy-and-paste error. The restrict-annotations policy isn't getting removed which is causing failures in the competing "allowed-annotations" test when it is run after this one.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Need to fix one error.

File: other/require-unique-external-dns/svc.yaml (hunk 1 above)
Comment (mintbomb27) [reply other/require-unique-external-dns/svc.yaml:5]: I'll change them 😅

File: other/restart-deployment-on-secret-change/06-check-restart.yaml (hunk 2 above)
Comment (mintbomb27) [reply other/restart-deployment-on-secret-change/06-check-restart.yaml:4]: Sounds like a better idea. I'll try this out.

Comment (mintbomb27) [reply other/restart-deployment-on-secret-change/06-check-restart.yaml:4]: So I'm not sure if the annotation is getting removed after the restart, I'm not able to get the annotation `ops.corp.com/trigger`. It's just isn't there. But I can see the annotation `deployment.kubernetes.io/revision` getting incremented everytime a restart happens. Should I check with that?

File: other/restrict-annotations/99-delete.yaml (hunk 3 above)
Comment (mintbomb27) [reply other/restrict-annotations/99-delete.yaml:6]: Sorry about this! Fixed it!

File: other/restart-deployment-on-secret-change/06-check-restart.yaml (hunk 2 above)
Comment (mintbomb27) [reply other/restart-deployment-on-secret-change/06-check-restart.yaml:4]: I still had to save the `resourceVersion` of the old secret in a text file and then take it to compare with the annotation. But it was fun! Realized that the `resourceVersion` gets updated only after the kyverno admission check.

**Review by chipzoller — APPROVED** [review]

**Pull request merged**